DOC-1227 ADD Example User_defined_header (#958)
tigergraph/server-docs: modules/data-loading/partials/load-part3B-specify-mapping.adoc

* `$0`, `$1`,...  refer to the first, second, ... columns in each line a data file.
* `SEPARATOR="|"` says the column separator character is the pipe (`|`). The default is comma (`,`).
* `HEADER="true"` says that the first line in the source contains column header names instead of data.  These names can be used instead of the columnn numbers.

[[user-defined-header-example]]
.Example using USER_DEFINED_HEADER

`USER_DEFINED_HEADER` specifies a header defined by `DEFINE HEADER`. The loader uses the user-defined column names instead of the column names in the input file.

.Loading job
[source,gsql]
----
# define the loading job
CREATE LOADING JOB load_simprod FOR GRAPH gsql_demo {
  DEFINE HEADER head1 = "id", "hash", "words";
  DEFINE FILENAME f1;

  LOAD f1
  TO VERTEX Product VALUES ($"id", $"hash"),
  TO TEMP_TABLE t (pid, description)
    VALUES ($"id", FLATTEN($"words", ",", 1))
  USING QUOTE="double",
        HEADER="true",
        USER_DEFINED_HEADER="head1";
}

# load the data
RUN LOADING JOB load_simprod USING f1="../simprod/data/simprod_data.csv";
----

> In this example, the input file contains the column names `ProductID`, `ImageHash`, and `Description`. The loading job references these columns using the user-defined names `id`, `hash`, and `words` specified by `DEFINE HEADER` because `USER_DEFINED_HEADER="head1"` is used.

Reviewer: ```suggestion
In this example, the input file contains the column names `ProductID`, `ImageHash`, and `Description` from left to right. The loading job overrides these names with `id`, `hash`, and `words` specified by `DEFINE HEADER` because `USER_DEFINED_HEADER="head1"` is used.
```
